### PR TITLE
Consume build common with target ES2020

### PR DESCRIPTION
**Pull request by jatgarg**
## Description

Item: https://dev.azure.com/fluidframework/internal/_workitems/edit/2772

Bump build common dependency with target ES2020 in tsconfig. We also needed to upgrade versions of "puppeteer" and "jest-puppeteer" as part of this.
Puppeteer is tied to a specific version of Chromium, and the version we were using were corresponding to an older browser that doesn't support all the es2020 constructs. So we needed to upgrade puppeteer.
Jest-puppeteer also needed to be upgraded as puppeteer was peer dependency and new version of puppeteer was not supported by present version of jest-puppeteer.

### CODE REVIEW

File: packages/common/core-interfaces/src/provider.ts [hunk 1]
```diff
@@ -22,7 +22,6 @@
  * This pattern enables discovery, and delegation in a standard way which is central
  * to FluidObject pattern
  *
- * @internal
```
Comment (jatgarg) [inline packages/common/core-interfaces/src/provider.ts:25]: This was causing build break. So removed it as it has been exported from the index.ts

File: api-report/core-interfaces.api.md [hunk 2]
```diff
@@ -4,8 +4,6 @@
 
 ```ts
 
-// Warning: (ae-incompatible-release-tags) The symbol "FluidObject" is marked as @public, but its signature references "FluidObjectProviderKeys" which is marked as @internal
```
Comment (jatgarg) [inline api-report/core-interfaces.api.md:7]: See comment above.

**Review by tylerbutler — APPROVED** [review]
Add a changeset since this is relevant to package consumers; other than that, looks great!

**Review by tylerbutler — APPROVED** [review]

**Comment by jatgarg** [comment]
@vladsud @christiango for improvements in bundle sizes.

**Comment by jatgarg** [comment]
<html>
<body>
<summary style="box-sizing: border-box; display: list-item; cursor: pointer;"><b style="box-sizing: border-box; font-weight: var(--base-text-weight-semibold, 600);"><span> </span>@fluid-example/bundle-size-tests:</b><span> </span>-75 KB</summary><slot id="details-content" style="color: rgb(31, 35, 40); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, &quot;Noto Sans&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(255, 255, 255); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;">

Metric Name | Baseline Size | Compare Size | Size Diff
-- | -- | -- | --
aqueduct.js | 448.4 KB | 429.92 KB | ⯆ -18.48 KB
connectionState.js | 680 Bytes | 680 Bytes | ■ No change
containerRuntime.js | 239.47 KB | 228.99 KB | ⯆ -10.49 KB
loader.js | 154.4 KB | 148.07 KB | ⯆ -6.33 KB
map.js | 46.66 KB | 44.81 KB | ⯆ -1.85 KB
matrix.js | 146.59 KB | 138.66 KB | ⯆ -7.93 KB
odspDriver.js | 92.31 KB | 87.04 KB | ⯆ -5.27 KB
odspPrefetchSnapshot.js | 43.64 KB | 40.77 KB | ⯆ -2.87 KB
sharedString.js | 163.26 KB | 154.52 KB | ⯆ -8.75 KB
sharedTree2.js | 279.92 KB | 270.03 KB | ⯆ -9.89 KB
Total Size | 1.73 MB | 1.66 MB | ⯆ -75 KB

</slot>
</body>
</html>

File: packages/common/core-interfaces/src/provider.ts (hunk 1 above)
Comment (alexvy86) [reply packages/common/core-interfaces/src/provider.ts:25]: Do you have an example of the error it was causing? Removing it shouldn't have any functional implications but it serves some purpose being there, so I want to understand how would it cause issues with the build.

**Pull request merged**